src/Microsoft.Identity.Web/CertificateManagement/CertificateDescription.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Description of a certificate.
+    /// </summary>
+    public class CertificateDescription
+    {
+        /// <summary>
+        /// Creates a certificate description from a certificate (by code).
+        /// </summary>
+        /// <param name="x509certificate2">Certificate.</param>
+        /// <returns>A certificate description.</returns>
+        public static CertificateDescription FromCertificate(X509Certificate2 x509certificate2)
+        {
+            return new CertificateDescription
+            {
+                Certificate = x509certificate2,
+            };
+        }
+
+        /// <summary>
+        /// Creates a Certificate Description from KeyVault.
+        /// </summary>
+        /// <param name="keyVaultUrl"></param>
+        /// <param name="certificateName"></param>
+        /// <param name="version"></param>
+        /// <returns>A certificate description.</returns>
+        public static CertificateDescription FromKeyVault(string keyVaultUrl, string certificateName, string version)
+        {
+            return new CertificateDescription
+            {
+                SourceType = CertificateSource.KeyVault,
+                Container = keyVaultUrl,
+                ReferenceOrValue = certificateName,
+            };
+            // todo support values?
```

Comment:
Do we want to keep this comments.
I guess, we could have an increment, if customers want it, to support policies: getting a cert from subject name, issuer, etc ...
Let's keep what we have and learn from customers/partners feedback.

Reply:
wasn't ready for review yet, but yes, have removed this comment.

---
In reply to: [436369394](https://github.com/AzureAD/microsoft-identity-web/pull/190#discussion_r436369394) [](ancestors = 436369394)